src/lib/dialog_ui.sh

```diff
@@ -108,6 +108,85 @@ function create_menu_options()
   run_dialog_command "$cmd" "$flag"
 }
 
+# Create menu with pagination capabilities. This function creates a screen similar
+# to the `create_menu_options`, but it allows the caller to partition the list of
+# options in pages.
+#
+# @box_title: Title of the screen
+# @message_box: Message box of the screen
+# @_options_array: An array reference containing menu options
+# @starting_index: Starting index of range of options to be displayed
+# @ending_index: Ending index of range of options to be displayed
+# @previous_label: Label of 'Previous' button.
+# @next_label: Label of 'Next' button.
+# @exit_label: Label of 'Exit' button.
+# @height: Menu height in lines size
+# @width: Menu width in column size
+# @flag How to display a command, the default value is
+#   "SILENT". For more options see `src/lib/kwlib.sh` function `cmd_manager`
+#
+# Return:
+# The index in `_options_array` of the option selected is stored in the `menu_return_string`
+# variable. If the user hit the 'OK' button 0 is returned, if hit the 'Exit' button 2
+# is returned, if hit the `Previous` button 3 is returned and if hit the `Next` button 1
+# is returned.
+function create_menu_options_with_pagination()
```

Comment:
Do we really need this function? This looks too specific to be part of a lib. Maybe I'm missing something but it looks like you can change `create_menu_options()` to make it more flexible to support the behavior that you implemented here.

Reply:
Additionally, after passing through this commit, I believe this change does not belong to this commit. You need to create a specific commit only for this.